reference/sql/statements/alter-index.md

```diff
@@ -0,0 +1,181 @@
+| t1    | CREATE TABLE `t1` (
+  `c1` int(11) DEFAULT NULL,
+  `c2` int(11) DEFAULT NULL,
+  UNIQUE KEY `c2` (`c2`),
+  UNIQUE KEY `c1` (`c1`) /*!80000 INVISIBLE */
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin |
++-------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+1 row in set (0.00 sec)
+```
+
+优化器将无法使用**不可见的索引**：
+
+{{< copyable "sql" >}}
+
+```sql
+EXPLAIN SELECT c1 FROM t1 ORDER BY c1; 
+```
+
+```
++-------------------------+----------+-----------+---------------+--------------------------------+
+| id                      | estRows  | task      | access object | operator info                  |
++-------------------------+----------+-----------+---------------+--------------------------------+
+| Sort_4                  | 10000.00 | root      |               | test.t1.c1:asc                 |
+| └─TableReader_8         | 10000.00 | root      |               | data:TableFullScan_7           |
+|   └─TableFullScan_7     | 10000.00 | cop[tikv] | table:t1      | keep order:false, stats:pseudo |
++-------------------------+----------+-----------+---------------+--------------------------------+
+3 rows in set (0.00 sec)
+```
+
+作为对比，c2 是**可见的索引**，优化器将可以使用索引：
+
+{{< copyable "sql" >}}
+
+```sql
+EXPLAIN SELECT c2 FROM t1 ORDER BY c2; 
+```
+
+```
++------------------------+----------+-----------+------------------------+-------------------------------+
+| id                     | estRows  | task      | access object          | operator info                 |
++------------------------+----------+-----------+------------------------+-------------------------------+
+| IndexReader_13         | 10000.00 | root      |                        | index:IndexFullScan_12        |
+| └─IndexFullScan_12     | 10000.00 | cop[tikv] | table:t1, index:c2(c2) | keep order:true, stats:pseudo |
++------------------------+----------+-----------+------------------------+-------------------------------+
+2 rows in set (0.00 sec)
+```
+
+即使用 SQL Hint `USE INDEX` 强制使用索引，优化器也无法使用不可见索引，这样的 SQL 语句会报错：
+
+{{< copyable "sql" >}}
+
+```sql
+SELECT * FROM t1 USE INDEX(c1);
+```
+
+```
+ERROR 1176 (42000): Key 'c1' doesn't exist in table 't1'
+```
+
+注意，“不可见”是仅仅对优化器而言的，不可见索引仍然可以被修改或删除。
+
+{{< copyable "sql" >}}
+
+```sql
+ALTER TABLE t1 DROP INDEX c1;
+```
+
+```
+Query OK, 0 rows affected (0.02 sec)
+```
+
+### 限制
+
+MySQL 对不可见索引有一条限制：不能将**主键**设置为不可见的。TiDB 兼容这条限制，这种情况会会抛出错误。
+
+{{< copyable "sql" >}}
+
+```sql
+CREATE TABLE t2(c1 INT, PRIMARY KEY(c1) INVISIBLE);
+```
+
+```
+ERROR 3522 (HY000): A primary key index cannot be invisible
+```
+
+这里的**主键**包括隐式的主键，隐式主键指的是当表中不存在显式的主键时，第一个 UNIQUE 索引（要求满足索引上的每一列都是 NOT NULL）会成为隐式的主键。
```

Comment:
please describe that implict PK doesnt exist in tidb actaully, but we treat it exists so that invisible index behaves the same with mysql

Reply:
@bb7133 addressed